### PR TITLE
Add dalex

### PR DESCRIPTION
**Pull request by hbaniecki**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [n/a] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hbaniecki** [comment]
@conda-forge-admin, please ping team
ready

File: recipes/dalex/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  sha256: 1a2f70259c4df7d1ae7c075d8b4a66a21586c0f9764a725cf367bc0d3a4b46f3
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (chrisburr) [inline recipes/dalex/meta.yaml:19]: Do you need a compiler? You either need to remove the compiler or remove `noarch: python` (as it needs to be platform dependent if you're compiling code).

If you don't need a compiler you should probably also:
* remove numpy from `host`
* Remove pin_compatible on numpy from `run` in favour of `>=1.18.4`
* Remove the >=3.6 on the Python dependencies in hosr and run

If you do need a compiler you should either:
* Remove the numpy version constraint in host so the global pin is used instead (1.16 currently)
* Or use `- numpy =1.18` for now (only if absolutely necessary, it'll cause a maintance burden to try and keep a compatible version).

Comment (hbaniecki) [reply recipes/dalex/meta.yaml]: Thanks, the overall `host` meaning wasn't clear to me, and if every package using `numpy` should have this. 

Why would I remove `python >=3.6`?

Comment (hbaniecki) [reply recipes/dalex/meta.yaml]: @chrisburr

Comment (chrisburr) [reply recipes/dalex/meta.yaml]: > Why would I remove python >=3.6?

If the package is **not** `noarch: python` it has to be built once per Python version per platform so it is currently built up to 5 times, once for each of `=3.6 cpython*`, `=3.7 cpython*`, `=3.8 cpython*`, `=3.9 cpython*` and `=3.6 pypy*`. Having `>=3.6` breaks this mechanism and `skip: true  # [py<36]` has to be used instead in the build section. If this package **is** `noarch: python` then none of this stuff is done and the version constraint (`>=3.6`) should be applied to `python` directly.

Comment (hbaniecki) [reply recipes/dalex/meta.yaml]: Thanks

**Pull request merged**